treeherder/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import division
+
+import json
+import logging
+import re
+import time
+from collections import Counter
+from datetime import (date,
+                      datetime,
+                      timedelta)
+
+import requests
+from django.conf import settings
+from django.db.models import Count
+from jinja2 import Template
+from requests.exceptions import RequestException
+from six import iteritems
+
+from treeherder.intermittents_commenter.constants import (COMPONENTS,
+                                                          TRIAGE_PARAMS,
+                                                          WHITEBOARD_NEEDSWORK_OWNER)
+from treeherder.model.models import (BugJobMap,
+                                     Push)
+from treeherder.webapp.api.utils import get_repository
+
+logger = logging.getLogger(__name__)
+
+
+class Commenter(object):
+    """handles fetching, transforming and submitting bug comments based on
+    daily or weekly thresholds and date range; if in dry_run, comments
+    will be output to stdout rather than submitted to bugzilla"""
+
+    def __init__(self, weekly_mode=False, dry_run=False):
+        self.weekly_mode = weekly_mode
+        self.dry_run = dry_run
+        self.session = self.new_request()
+
+    def run(self):
+        startday, endday = self.calculate_date_strings(self.weekly_mode, 6)
+        alt_startday, alt_endday = self.calculate_date_strings(True, DISABLE_DAYS=21)
+        all_params = self.create_comments(startday, endday, alt_startday, alt_endday)
```

Comment:
I think `all_params` is a bit unclear here. Really it's a list of comments, yeah? So perhaps `comments`? (And similarly for the argument name in `print_or_submit_comments()`.)

Reply:
hrmm.. technically, its a list of dicts containing bug_id, comment, and potentially new whiteboard and priority text. I could change it to 'comments' for simplicity's sake, unless you have other ideas?

Reply:
Ah true. The non-comment bug changes were a more recent addition to the old commenter.

Maybe `bug_changes`? (And I suppose `print_or_submit_comments()` and `create_comments()` updated to reflect their additional functionality?)